core/src/main/scala/cats/ApplicativeError.scala

```diff
@@ -216,6 +215,12 @@ trait ApplicativeError[F[_], E] extends Applicative[F] {
       case NonFatal(e) => raiseError(e)
     }
 
+  /**
+   * Evaluates the specified block, catching exceptions of the specified type. Uncaught exceptions are propagated.
+   */
+  def catchOnly[T >: Null <: Throwable]: CatchOnlyPartiallyApplied[T, F, E] =
+    new CatchOnlyPartiallyApplied[T, F, E](this)
```

Comment:
Out of curiosity... I'm struggling to understand this.

What was the `T >: Null` constraint added for in there? To me it seems that due to the latter constraint `T <: Throwable` the former one will be always satisfied automatically, won't it? This is because `Throwable` is `AnyRef` and `Null <: T <: AnyRef` for any `T`.

Or am I missing something?
@takayahilton could you clarify please?